lib/src/main/java/scal/io/liger/MainActivity.java

```diff
@@ -589,14 +602,26 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
 
                 if (c instanceof ClipCard) {
                     ClipCard cc = (ClipCard)c;
-                    MediaFile mf = new MediaFile(path, Constants.PHOTO);
-                    cc.saveMediaFile(mf);
 
-                    // SEEMS LIKE A REASONABLE TIME TO SAVE
-                    mStoryPathLibrary.save(true);
+                    // confirm mime type
+                    String mimeType = URLConnection.guessContentTypeFromName(path);
+
+                    Log.d(TAG, "onActivityResult, media type is " + mimeType);
+
+                    if (mimeType.startsWith(Constants.IMAGE)) {
 
-                    mCardAdapter.changeCard(cc);
-                    scrollRecyclerViewToCard(cc);
+                        MediaFile mf = new MediaFile(path, Constants.PHOTO); // TODO: switch over to "image" to match mime type?
```

Comment:
I dont think we want to do that, might as well track the user visible name for medium, which is Photo